### DISCUSSION TITLE
`TooltipReference` ref being too strict [TS]

### DISCUSSION THREAD

**Discussion by matheus1lva** [ariakit/ariakit#833] [Questions] [answered]
Hello!
I'm working with the tooltip, i'm trying to build my own as stated on the abstract example:

```
import React from "react";
import {
  useTooltipState,
  Tooltip as ReakitTooltip,
  TooltipReference,
} from "reakit/Tooltip";

function Tooltip({ children, title, ...props }) {
  const tooltip = useTooltipState();
  return (
    <>
      <TooltipReference {...tooltip} ref={children.ref} {...children.props}>
        {(referenceProps) => React.cloneElement(children, referenceProps)}
      </TooltipReference>
      <ReakitTooltip {...tooltip} {...props}>
        {title}
      </ReakitTooltip>
    </>
  );
}

function Example() {
  return (
    <Tooltip title="Tooltip">
      <button>Reference</button>
    </Tooltip>
  );
}
```

Whenever I try to add anything to the `ref` props, it complains that it has to be of type `HTMLDivElement` even though on the example we can use a button...

Maybe the typing is wrong?

**Comment by diegohaz** [comment]
That doesn't seem correct indeed. Could you please create a TypeScript CodeSandbox so we can play around with that?

**Reply by matheus1lva** [reply]
https://codesandbox.io/s/reakit-tooltip-tuli1

**Comment by diegohaz** [comment, marked as answer]
It looks like you just need to type your children prop correctly: https://codesandbox.io/s/reakit-tooltip-forked-t7rgn?file=/src/App.tsx

**Reply by matheus1lva** [reply]
thank you!
